<launch>

    <!-- defines name of the namespace of the drone -->
    <arg name="UAV_NAME" default="$(env UAV_NAME)" />
    <arg name="RUN_TYPE" default="$(env RUN_TYPE)" />
  
    <arg name="simulation" default="true" />
  
    <!-- will it run using GNU debugger? -->
    <arg name="DEBUG" default="false" />
    <arg unless="$(arg DEBUG)" name="launch_prefix_debug" value=""/>
    <arg     if="$(arg DEBUG)" name="launch_prefix_debug" value="debug_roslaunch"/>
  
    <!-- will it run as standalone nodelet or using a nodelet manager? -->
    <arg name="standalone" default="true" />
    <arg name="manager" default="$(arg UAV_NAME)_Ros_Pub_Sub" />
    <arg unless="$(arg standalone)" name="nodelet" value="load"/>
    <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
    <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
    <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>
  
    <group ns="$(arg UAV_NAME)">
        <node pkg="nodelet" type="nodelet" name="Ros_Pub" args="$(arg nodelet) ros_pub/Ros_Pub $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">
    
              <!-- ROS parameters config file -->
              <param name="uav_name" value="$(arg UAV_NAME)" />
              <param name="target_x" value="$(arg X)" />
              <param name="target_y" value="$(arg Y)" />
              <param name="target_z" value="$(arg Z)" />
  
              <!-- Subscribers (note that the topic names will also be remapped using the namespace as a prefix) -->
              <remap from="~odom_uav_in" to="odometry/odom_main"/>
  
              <!-- Publishers (note that the topic names will also be remapped using the namespace as a prefix) -->
              <remap from="~dist_to_target_out" to="~dist_to_target" />
          
          </node>
    </group>
    
    <group ns="$(arg UAV_NAME)">
        <node pkg="nodelet" type="nodelet" name="Ros_Sub" args="$(arg nodelet) ros_sub/Ros_Sub $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">
    
            <!-- ROS parameters config file -->
            <param name="uav_name" value="$(arg UAV_NAME)" />
  
            <!-- Subscribers (note that the topic names will also be remapped using the namespace as a prefix) -->
            <remap from="~dist_to_target_in" to="Ros_Pub/dist_to_target"/>
        
        </node>
    </group>

  </launch>